cpp/ql/lib/experimental/quantum/Language.qll

```diff
@@ -93,7 +93,10 @@ module GenericDataSourceFlow = TaintTracking::Global<GenericDataSourceFlowConfig
 
 private class ConstantDataSource extends Crypto::GenericConstantSourceInstance instanceof OpenSslGenericSourceCandidateLiteral
 {
-  override DataFlow::Node getOutputNode() { result.asExpr() = this }
+  override DataFlow::Node getOutputNode() {
+    // A literal can be a string or an int, so handling both indirect and direct cases
```

Comment:
Could you please elaborate on the two cases and why `asIndirectExpr()` is needed? At first glance, strings don't require this pattern, so if we need to revisit this, the intent or a test case is important.

Reply:
Clarified this myself. Could you please revise the comment to be something like the following?
```
    // OpenSSL algorithms may be referenced either by string name or by numeric ID:
    // String names (e.g. "AES-256-CBC") appear in the AST as character pointer
    // literals. For these we must use `asIndirectExpr`. Numeric IDs (e.g. NID_aes_256_cbc)
    // appear as integer literals. For these, we must use `asExpr` to get the "value" node.
    [result.asIndirectExpr(), result.asExpr()] = this
```